### PR TITLE
Sanitize authentication headers

### PR DESCRIPTION
**Pull request by patryk**

### CODE REVIEW

**Review by manatarms — APPROVED** [review]

File: src/API/AbstractAPIClient.php [hunk 1]
```diff
@@ -78,6 +78,17 @@ public function callAPI(Request $request)
...
     }
 
+    /**
+     * @param Request $request
+     * @return Request
+     */
+    public function sanitizeHeaders(Request $request)
+    {
+        $request->removeHeader('Authorization');
+        $request->removeHeader('X-Auth-Key');
```
Comment (mgirouard) [inline src/API/AbstractAPIClient.php:89]: I think auth email may also need to be here too.

It's PII and, given most installs will be on shared servers, it's probably best to include that with this patch.

Comment (mgirouard) [reply src/API/AbstractAPIClient.php:89]: Actually, I think we can just remove the call entirely. We're double-logging.

**Review by mgirouard — CHANGES_REQUESTED** [review]

File: src/API/AbstractAPIClient.php (hunk 1 above)
Comment (patryk) [reply src/API/AbstractAPIClient.php:89]: Removed `X-Auth-Email`.

**Comment by jwineman** [comment]
I just want to say that @dfritsch told me to write it like that, this bug is his fault!

**Review by mgirouard — APPROVED** [review]

**Review by hunts — APPROVED** [review]

**Pull request merged**